controllers/watcher_controller.go

```diff
@@ -69,6 +69,12 @@ func (r *WatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	logger := ctrlLog.FromContext(ctx).WithName(req.NamespacedName.String())
 	logger.Info("Reconciliation loop starting")
 
+	// sync CA Certificate to Istio Namespace
+	if err := r.IstioClient.SyncCertificateSecretToIstio(ctx, r.Client); err != nil {
```

Comment:
why do you need to pass a client into a method of a client? This looks a bit suspicious. Maybe you want to refactor the structure of these clients / methods